### PR TITLE
Bitwise_and operator

### PR DESCRIPTION
**Pull request by CharlieL7**
* Implements the bitwise_and operator and ONNX parser.
* Needed for support in TorchMIGraphX models

### CODE REVIEW

**Review by umangyadav — COMMENTED** [review]
Looks good. Fix tidy errors

File: src/include/migraphx/op/bitwise_and.hpp [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        }
+        return result;
+    }
+
+    auto apply() const
+    {
+        return [](auto x, auto y) {
+            if constexpr(std::is_integral<decltype(x)>{} and std::is_integral<decltype(y)>{})
+            {
+                return x & y;
```
Comment (TedThemistokleous) [inline src/include/migraphx/op/bitwise_and.hpp:61]: add a // NOLINT here with a comment

File: test/ref/bitwise_and.cpp [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    p.compile(migraphx::make_target("ref"));
+    auto result = p.eval({}).back();
+    std::vector<int32_t> results_vector;
+    result.visit([&](auto output) { results_vector.assign(output.begin(), output.end()); });
+    std::vector<int32_t> gold(data2.size());
+    std::transform(data1.begin(),
+                   data1.end(),
+                   data2.begin(),
+                   gold.begin(),
+                   [](int32_t n1, int32_t n2) -> int32_t { return n1 & n2; });
```
Comment (TedThemistokleous) [inline test/ref/bitwise_and.cpp:52]: Add // NOLINT here too

File: test/ref/bitwise_and.cpp [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    params0["r"] = migraphx::argument(input_fixed_shape0, right_data.data());
+    auto result  = p.eval(params0).back();
+    std::vector<char> results_vector;
+    result.visit([&](auto output) { results_vector.assign(output.begin(), output.end()); });
+    std::vector<uint8_t> gold(left_data.size());
+    std::transform(left_data.begin(),
+                   left_data.end(),
+                   right_data.begin(),
+                   gold.begin(),
+                   [](uint8_t n1, uint8_t n2) -> uint8_t { return n1 and n2; });
```
Comment (TedThemistokleous) [inline test/ref/bitwise_and.cpp:82]: and -> &

**Review by TedThemistokleous — APPROVED** [review]
Looks good over all. Fix some of the tidy/lint warnings and that's really it.

**Review by TedThemistokleous — COMMENTED** [review]
Fix Errors. that I'll approve.

**Comment by causten** [comment]
CI Hit a failure, not sure why it is not showing up....

[2024-07-30T21:21:39.312Z] [   RUN    ] test_bitwise_and<migraphx::shape::bool_type>
[2024-07-30T21:21:39.312Z] 
[2024-07-30T21:21:39.312Z] module: "main"
[2024-07-30T21:21:39.312Z] y = @param:y -> bool_type, {8}, {1}
[2024-07-30T21:21:39.312Z] x = @param:x -> bool_type, {8}, {1}
[2024-07-30T21:21:39.312Z] @2 = convert[target_type=2](x) -> float_type, {8}, {1}
[2024-07-30T21:21:39.312Z] @3 = convert[target_type=2](y) -> float_type, {8}, {1}
[2024-07-30T21:21:39.312Z] @4 = bitwise_and(x,y) -> bool_type, {8}, {1}
[2024-07-30T21:21:39.312Z] 
[2024-07-30T21:21:39.312Z] 
[2024-07-30T21:21:39.312Z] FAILED: test_bitwise_and<migraphx::shape::bool_type>
[2024-07-30T21:21:39.312Z]     what(): /var/jenkins/workspace/AMDMIGraphX_PR-3321/src/include/migraphx/op/bitwise_and.hpp:51: compute_shape: BITWISE_AND: only supports integral types

**Comment by CharlieL7** [comment]
> CI Hit a failure, not sure why it is not showing up....
> 
> [2024-07-30T21:21:39.312Z] [ RUN ] test_bitwise_andmigraphx::shape::bool_type [2024-07-30T21:21:39.312Z] [2024-07-30T21:21:39.312Z] module: "main" [2024-07-30T21:21:39.312Z] y = @param:y -> bool_type, {8}, {1} [2024-07-30T21:21:39.312Z] x = @param:x -> bool_type, {8}, {1} [2024-07-30T21:21:39.312Z] @2 = convert[target_type=2](x) -> float_type, {8}, {1} [2024-07-30T21:21:39.312Z] @3 = convert[target_type=2](y) -> float_type, {8}, {1} [2024-07-30T21:21:39.312Z] @4 = bitwise_and(x,y) -> bool_type, {8}, {1} [2024-07-30T21:21:39.312Z] [2024-07-30T21:21:39.312Z] [2024-07-30T21:21:39.312Z] FAILED: test_bitwise_andmigraphx::shape::bool_type [2024-07-30T21:21:39.312Z] what(): /var/jenkins/workspace/AMDMIGraphX_PR-3321/src/include/migraphx/op/bitwise_and.hpp:51: compute_shape: BITWISE_AND: only supports integral types

Should not hit this error anymore, removed the bool type test since it's not supported

**Pull request merged**